### PR TITLE
Delete before a given date

### PR DESCRIPTION
**Pull request by acgrobman**

### CODE REVIEW

File: public/locales/en/administration.json [hunk 1]
```diff
@@ -158,6 +158,8 @@
             "uploaded": "I have uploaded the downloaded data to a safe file storage location.",
             "irreversible": "I understand this process is irreversible and the deleted data cannot be recovered from this system."
         },
+        "date": "Date to Delete Data Before",
+        "tooRecent": "This date is too recent. For technical reasons, you must choose a date at least a week in the past.",
         "submit": "Delete"
```
Comment (CarsonHerness) [inline public/locales/en/administration.json:163]: You're missing the error value for invalid dates: `downloadData.error.invalid`

File: public/locales/es/administration.json [hunk 2]
```diff
@@ -158,6 +158,8 @@
             "uploaded": "He almacenado los datos descargados en un lugar seguro.",
             "irreversible": "Entiendo que este proceso es irreversible y no se puede recuperar los datos borrado."
         },
+        "date": "Borrar datos más viejos que esta fecha",
+        "tooRecent": "Esta fecha es tan recientemente. Debido a razones técnicas, tienes que elegir una fecha más de una semana en el pasado.",
```
Comment (CarsonHerness) [inline public/locales/es/administration.json:162]: Same missing value: `downloadData.error.invalid`

**Review by CarsonHerness — APPROVED** [review]
Looks mostly good, just a missing error

File: src/components/Admin/ManageSensors/DeleteOldDataModal.tsx [hunk 3]
```diff
@@ -55,14 +68,55 @@ function DeleteOldDataModal(): JSX.Element {
...
+    const date = new Date(year, month - 1, day);
+    /**
+     * Determines whether a given date is old enough to serve as the delete before date.
+     *
+     * @returns Whether the inputted date is old enough to be the delete before date
+     */
+    const oldEnough = () => {
+      const offset = 7;
+      const sevenDaysAgo = new Date();
+      sevenDaysAgo.setDate(sevenDaysAgo.getDate() - offset);
```
Comment (akampen) [inline src/components/Admin/ManageSensors/DeleteOldDataModal.tsx:93]: does this work as desired in the beginning of the month (i.e. on the 1st through 6th)?

Comment (acgrobman) [reply src/components/Admin/ManageSensors/DeleteOldDataModal.tsx:93]: It should per the below screenshot (ignore me not knowing how to do math, the point still stands despite the variable names)
image

**Review by akampen — APPROVED** [review]

**Pull request merged**